main/openrc/firstboot.initd

```diff
@@ -0,0 +1,44 @@
+#!/sbin/openrc-run
+
+# The first boot init service
+
+# read kernel options
+init_KOPT() {
+	for opt in $(cat /proc/cmdline 2>/dev/null); do
+		case "$opt" in
+			ssh_*=*)
+				eval "KOPT_${opt%%=*}='${opt#*=}'" ;;
+		esac
+	done
+}
+
+depend() {
+	keyword -vserver -lxc
+	init_KOPT
+	if [ -n "$KOPT_ssh_key" ] || [ -n "$KOPT_ssh_pass" ]; then
+		need sshd
+	fi
+}
+
+
+start() {
+	init_KOPT
+	local rc=0
+	ebegin "Starting ${RC_SVCNAME}"
+	if [ -n "$KOPT_ssh_key" ] && [ ! -f "/root/.ssh/authorized_keys" ]; then
+		einfo "Fetching ssh keys"
+		mkdir -p /root/.ssh
```

Comment:
Should we be setting the mode on these files? OpenBSD uses 0700 for "/root/.ssh", and 0600 for "/root/.ssh/authorized_keys"

Reply:
Updated. Thanks.